shell/edit/monitoring.coreos.com.receiver/index.vue

```diff
@@ -43,6 +44,18 @@ export default {
   data() {
     this.$set(this.value, 'spec', this.value.spec || {});
 
+    if (this.mode === _EDIT || this.mode === _VIEW) {
```

Comment:
The logic for this feels like it should be in one place - some is here, the rest is in the model.

I suggest using before and after save hooks here, so that it is all in this file.

As is, if there is an error saving, I think it will break, as the model will clean up port and host, but the view will still expect them to be present.